### PR TITLE
Fixes CIM deserialization bug

### PR DESCRIPTION
**Pull request by PaulHigin**
This fix is ported from our internal branch and is related to TFS bug 11736179.

### CODE REVIEW

**Review by adityapatwardhan — APPROVED** [review]
LGTM

**Review by daxian-dbw — APPROVED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Utility/clixml.tests.ps1 [hunk 1]
```diff
@@ -185,3 +185,52 @@
...
+        # Only run on Windows platform.
+        # Ensure calc.exe is avaiable for test.
+        if ( !$IsWindows -or ((Get-Command calc.exe 2>$null) -eq $null) )
+        {
+            $orginalDefaultParameters = $PSDefaultParameterValues.Clone()
+            $PSDefaultParameterValues["it:skip"] = $true
+        }
+        else
+        {
+            (Get-Process -Name 'win32calc','calculator' 2>$null) | Stop-Process -force -ErrorAction SilentlyContinue
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/clixml.tests.ps1]: Minor change `-force ` -> `-Force`

File: test/powershell/Modules/Microsoft.PowerShell.Utility/clixml.tests.ps1 [hunk 2]
```diff
@@ -185,3 +185,52 @@
...
+
+    AfterAll {
+
+        if ($orginalDefaultParameters -ne $null)
+        {
+            $PSDefaultParameterValues = $orginalDefaultParameters
+        }
+        else
+        {
+            (Get-Process -Name 'win32calc','calculator' 2>$null) | Stop-Process -force -ErrorAction SilentlyContinue
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/clixml.tests.ps1]: The same.

File: src/System.Management.Automation/engine/serialization.cs [hunk 3]
```diff
@@ -2979,6 +2979,35 @@ internal InternalDeserializer(XmlReader reader, DeserializationContext context)
...
+        #region Known CIMTypes
+
+        private static Lazy<HashSet<Type>> s_knownCimArrayTypes = new Lazy<HashSet<Type>>(
+            () =>
+                new HashSet<Type>
+                {
+                    typeof(Boolean),
+                    typeof(byte),
+                    typeof(char),
+                    typeof(DateTime),
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/serialization.cs]: Do we use `DateTimeOffset` ?

**Review by iSazonov — COMMENTED** [review]
Leave a comment

File: test/powershell/Modules/Microsoft.PowerShell.Utility/clixml.tests.ps1 (hunk 1 above)
Comment (PaulHigin) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/clixml.tests.ps1]: Fixed

File: test/powershell/Modules/Microsoft.PowerShell.Utility/clixml.tests.ps1 (hunk 2 above)
Comment (PaulHigin) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/clixml.tests.ps1]: Fixed

File: src/System.Management.Automation/engine/serialization.cs (hunk 3 above)
Comment (PaulHigin) [inline src/System.Management.Automation/engine/serialization.cs]: No

Comment (iSazonov) [inline src/System.Management.Automation/engine/serialization.cs]: Thanks!

**Comment by daxian-dbw** [comment]
The new commit is a simple letter case change. It's OK to use the existing approvals.

**Pull request merged**